raphtory-graphql/src/server.rs

```diff
@@ -58,31 +63,19 @@ pub struct RaphtoryServer {
     data: Data,
 }
 
-// Define a struct for log configuration
-#[derive(Debug, Serialize, Deserialize)]
-pub struct LogConfig {
-    log_level: String,
-}
-
 impl RaphtoryServer {
-    /// Return a server object with graphs loaded from a map `graphs`
-    pub fn from_map(graphs: HashMap<String, MaterializedGraph>) -> Self {
-        let data = Data::from_map(graphs);
-        Self { data }
-    }
-
-    /// Return a server object with graphs loaded from a directory `graph_directory`
-    pub fn from_directory(graph_directory: &str) -> Self {
-        let data = Data::from_directory(graph_directory);
-        Self { data }
-    }
-
-    /// Return a server object with graphs loaded from a map `graphs` and a directory `graph_directory`
-    pub fn from_map_and_directory(
-        graphs: HashMap<String, MaterializedGraph>,
-        graph_directory: &str,
+    pub fn new(
+        work_dir: &Path,
```

Comment:
We might consider making `work_dir` optional and default to something like /tmp/raphtory/graph-cache. Also, now that we write the graphs into files for the cache, maybe having the option of loading graphs from a directory doesn't make sense anymore. We could remove that option altogether and make `graphs` mandatory.

Reply:
Actually, having a list of paths makes sense to avoid loading all the graphs into memory at once, so I take that back 🤐